packages/vm/src/evm/opcodes/util.ts

```diff
@@ -231,30 +234,25 @@ export function writeCallOutput(runState: RunState, outOffset: BN, outLength: BN
  * @param {Buffer}   currentStorage
  * @param {Buffer}   value
  * @param {Buffer}   keyBuf
+ * @param {Common}   common
  */
 export function updateSstoreGas(
   runState: RunState,
   currentStorage: Buffer,
   value: Buffer,
-  keyBuf: Buffer,
   common: Common
-) {
-  const sstoreResetCost = common.param('gasPrices', 'sstoreReset')
+): BN {
   if (
     (value.length === 0 && !currentStorage.length) ||
     (value.length !== 0 && currentStorage.length)
   ) {
-    runState.eei.useGas(
-      new BN(adjustSstoreGasEIP2929(runState, keyBuf, sstoreResetCost, 'reset', common)),
-      'updateSstoreGas'
-    )
+    const gas = new BN(common.param('gasPrices', 'sstoreReset'))
+    return gas
   } else if (value.length === 0 && currentStorage.length) {
-    runState.eei.useGas(
-      new BN(adjustSstoreGasEIP2929(runState, keyBuf, sstoreResetCost, 'reset', common)),
-      'updateSstoreGas'
-    )
+    const gas = new BN(common.param('gasPrices', 'sstoreReset'))
     runState.eei.refundGas(new BN(common.param('gasPrices', 'sstoreRefund')), 'updateSstoreGas')
-  } else if (value.length !== 0 && !currentStorage.length) {
-    runState.eei.useGas(new BN(common.param('gasPrices', 'sstoreSet')), 'updateSstoreGas')
+    return gas
+  } /*(value.length !== 0 && !currentStorage.length)*/ else {
```

Comment:
I commented this; it was an `else if` before. The situation reduces to either the value = 0, storage = 0 or value > 0, storage > 0 (first if statement), or value = 0, storage > 0, or (thus) value > 0, storage = 0. Thus the final statement can be reduced to an `else` statement, as all the other conditions are not true, thus this statement must be true.

Reply:
great, how about we add your reasoning above as a comment under the else statement so it's easy to follow?